src/main/java/gift/repository/ProductRepository.java

```diff
@@ -2,45 +2,89 @@
 
 import gift.entity.Product;
 
-import java.util.ArrayList;
+import java.sql.ResultSet;
+import java.sql.SQLException;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import org.springframework.jdbc.core.simple.JdbcClient;
 import org.springframework.stereotype.Repository;
-
+import org.springframework.jdbc.core.RowMapper;
 
 @Repository
 public class ProductRepository {
 
+    private final JdbcClient jdbcClient;
+
+    public ProductRepository(JdbcClient jdbcClient) {
+        this.jdbcClient = jdbcClient;
+    }
+
 
     private final Map<Long, Product> store = new HashMap<>();
     private Long nextId = 1L;
 
+
     public List<Product> findAll() {
-        return new ArrayList<>(store.values());
+        return jdbcClient.sql("SELECT * FROM product")
+                .query(new RowMapper<Product>() {
+                    @Override
+                    public Product mapRow(ResultSet rs, int rowNum) throws SQLException {
+                        return new Product(
+                                rs.getLong("id"),
+                                rs.getString("name"),
+                                rs.getInt("price"),
+                                rs.getString("image_url")
+                        );
+                    }
+                })
+                .list();
```

Comment:
```suggestion
return jdbcClient.sql("SELECT * FROM product")
                .query((rs, rowNum) -> new Product(
                        rs.getLong("id"),
                        rs.getString("name"),
                        rs.getInt("price"),
                        rs.getString("image_url")
                ))
                .list();
```

람다를 활용하면 좀 더 간단하게 표현이 가능합니다 😊